config/externalname.go

```diff
@@ -1037,6 +1037,9 @@ var terraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
 	"google_tags_tag_key": config.IdentifierFromProvider,
 	// Imported by using the following tagValues/{{name}}
 	"google_tags_tag_value": config.IdentifierFromProvider,
+
+	// cloudquotas
+	"google_cloud_quotas_quota_preference": config.TemplatedStringAsIdentifier("name", "{{ .parameters.parent }}/locations/global/quotaPreferences/{{ .external_name }}"),
```

Comment:
I think I got this right? https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_quotas_quota_preference#name-4  looks like a pretty cut and dry version of https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#gcp

Reply:
Yes, it seems okay, but since we are using the name parameter in the external name here, it would be a good idea to mark it as required. Here is an example configuration: https://github.com/crossplane-contrib/provider-upjet-gcp/blob/ae12d1217c86f7a88ee6072b59588d39267c09b2/config/redis/config.go#L28

Also, it would be good to move the `cloudquotas` group to the lines with cloud groups in alphabetical order.

Reply:
Done. I believe :) See 190ed4b722ec68cf5affce3279ec7136013712cf.

I was anticipating that adding this would affect the code generation and, in turn, mark it as required in the openapi schema at https://github.com/jacobstr/provider-upjet-gcp/blob/1db2f529d96c09b7de407c4775a9d1c3a8440a78/package/crds/cloudquotas.gcp.upbound.io_quotapreferences.yaml#L232